src/main/java/com/commercetools/sync/services/impl/BaseTransformServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.commercetools.sync.services.impl;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import com.commercetools.sync.commons.models.GraphQlQueryResources;
+import com.commercetools.sync.commons.models.ResourceIdsGraphQlRequest;
+import com.commercetools.sync.commons.models.ResourceKeyId;
+import com.commercetools.sync.commons.utils.ChunkUtils;
+import io.sphere.sdk.client.SphereClient;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+
+public class BaseTransformServiceImpl {
+  /*
+   * An id is a 36 characters long string. (i.e: 53c4a8b4-754f-4b95-b6f2-3e1e70e3d0c3) We
+   * chunk them in 300 ids, we will have a query around 11.000 characters. Above this size it
+   * could return - Error 413 (Request Entity Too Large)
+   */
+  public static final int CHUNK_SIZE = 300;
+  protected final Map<String, String> referenceIdToKeyCache;
+
+  private final SphereClient ctpClient;
+
+  protected BaseTransformServiceImpl(
+      @Nonnull final SphereClient ctpClient,
+      @Nonnull final Map<String, String> referenceIdToKeyCache) {
+    this.ctpClient = ctpClient;
+    this.referenceIdToKeyCache = referenceIdToKeyCache;
+  }
+
+  protected SphereClient getCtpClient() {
+    return ctpClient;
+  }
+
+  //  @SuppressFBWarnings(
```

Comment:
Is this comment required?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/697/commits/db6bca046d69046ac3d92b2b3bac89e65ac5f1f4